src/msxswagger/templates/swagger-sso-redirect.html

```diff
@@ -0,0 +1,125 @@
+<!doctype html>
+<html lang="en-US">
+<head><title>SSO Redirect</title></head>
+<body onload="run()">
+</body>
+</html>
+<script>
+    'use strict';
+    var InProgressRedirectKey = "CurrentRedirectSSO";
+    var SsoStateKeyPrefix = "sso-";
+
+    function run () {
+        var qp, arr;
+
+        if (/code|token|error/.test(window.location.hash)) {
+            qp = window.location.hash.substring(1);
+        } else {
+            qp = location.search.substring(1);
+        }
+
+        arr = qp.split("&")
+        arr.forEach(function (v,i,_arr) { _arr[i] = '"' + v.replace('=', '":"') + '"';})
+        qp = qp ? JSON.parse('{' + arr.join() + '}',
+            function (key, value) {
+                return key === "" ? value : decodeURIComponent(value)
+            }
+        ) : {}
+
+        var redirectCtx = loadRedirectContext({});
+        var sentState = redirectCtx.state;
+
+        var result = qp;
+        result.redirectUrl = redirectCtx.redirectUrl;
+        result.isValid = qp.state === sentState;
+
+        if (!result.isValid) {
+            result.error = {
+                authId: "SSO",
+                source: "SSO",
+                level: "error",
+                message: "Authorization may be unsafe, passed state was changed in server Passed state wasn't returned from auth server"
+            };
+            handleError({redirectCtx, state: qp.state, result})
+        } else if (result.code) {
+            handleSuccess({redirectCtx, state: qp.state, result})
+        } else {
+            var oauthErrorMsg
+            if (qp.error) {
```

Comment:
shouldn’t this have been `result.error`, for consistency?

It would have been nice, if we can find a way to pass the error object only, instead of piggybacking on `qp` (which is the query parameter) -- we are assigning more responsibility for the qp object than it needs to have.